docs/control-structures.rst

```diff
@@ -52,40 +57,53 @@ write::
 
 The names of output parameters can be omitted.
 The output values can also be specified using ``return`` statements.
-The ``return`` statements are also capable of returning multiple
-values, see :ref:`multi-return`.
+
+The ``return`` statements are also capable of returning multiple values using ``return (v0, v1, ..., vn)``. The number of parameters must equal the number of output parameters and they must all be the same type.
+
 Return parameters are initialized to zero; if they are not explicitly
 set, they stay to be zero.
 
 Input parameters and output parameters can be used as expressions in
 the function body.  There, they are also usable in the left-hand side
 of assignment.
 
+.. note::
+    You cannot return some types from non-internal functions. If you enable the
```

Comment:
Perhaps add: Notable exceptions are multi-dimensional dynamic arrays and structs.